napari/layers/base/base.py

```diff
@@ -698,12 +700,14 @@ def _get_extent_world(self, data_extent):
     @property
     def extent(self) -> Extent:
         """Extent of layer in data and world coordinates."""
-        data = self._extent_data
-        return Extent(
-            data=data,
-            world=self._get_extent_world(data),
-            step=abs(self._data_to_world.scale),
-        )
+        extent_data = self._extent_data
```

Comment:
doesn't have to be done here, but we really ought to use `functools.cached_property`, (and vendor it from functools for py<3.8).  This is exactly what it does, with a slightly cleaner API for invalidating the cache

Reply:
actually, it also would mean we don't need `self._extent_cache = None` above.  which is a bit nicer.  @Czaki would you want to do that here?  Or prefer to leave it for elsewhere

Reply:
But how did you would like to invalidate `functools.cached_property` on the update of property like `scale`, `translate` or updating data? 

`self._extent_cache` is settled to None on `_update_dims` call.

Reply:
is the question how does one invalidate a `cached_property`?  If so, you do it by deleting the attribute:
 https://docs.python.org/3/library/functools.html#functools.cached_property

so `self._extent_cache = None` would be the same as `del self.extent`

Reply:
I miss this sentence when read bout` cache_property`. 

When is a plan to drop python 3.7?

Reply:
end of the year... but that really doesn't need to come into play in our decision here. It's a tiny little bit of code to vendor

Reply:
I support vendor - I am not at my laptop for a while, so if someone wants to do this, that’s fine by me, otherwise I can do it tonight

Reply:
nice, I was not aware of `functools.cached_property`. @jni: we should keep this in mind as a future update to regionprops caching for scikit-image

Reply:
now using `cached_property`